File: 26-Dynamic-Programming-Part-II/cpp/05-super-egg-drop.cpp
#include <vector>

using namespace std;

class Solution {
public:
    int eggDrop(int n, int k) {
        vector<int> dp(n + 1, 0);
        int moves = 0;
        while (dp[n] < k) {
            moves++;
            for (int eggs = n; eggs >= 1; eggs--) {
                dp[eggs] = dp[eggs] + dp[eggs - 1] + 1;
            }
        }
        return moves;
    }
};

File: 26-Dynamic-Programming-Part-II/python/05_super_egg_drop.py
class Solution:
    def eggDrop(self, n: int, k: int) -> int:
        dp = [0] * (n + 1)
        moves = 0
        while dp[n] < k:
            moves += 1
            for eggs in range(n, 0, -1):
                dp[eggs] = dp[eggs] + dp[eggs - 1] + 1
        return moves
